### ISSUE TITLE
ScrollView preferred size doesn't apply equaly to the parent on all style

### ISSUE THREAD

**Issue by wuanzhuan** [slint-ui/slint#4503]
the popupwindow's parent element is in the red border
![image](https://github.com/slint-ui/slint/assets/11628049/4473f273-497f-4b3a-b883-ce71a476c582)
![image](https://github.com/slint-ui/slint/assets/11628049/4fcb8b78-424c-451b-8894-aa2d2e003756)

**Comment by wuanzhuan** [comment]
it is right when i remove the scrollview 
![image](https://github.com/slint-ui/slint/assets/11628049/76c5174e-f18c-405f-b190-2438bf258fab)
    popup := PopupWindow {
        x: 12px;
        y: parent.height + 12px;
        
        Rectangle {
            background: Palette.background;
            border-color: Palette.border;
            border-width: 1phx;  
            preferred-width: text-element.preferred-width + 12px;
            preferred-height: text-element.preferred-height + 12px;
            text-element := HorizontalLayout {
                padding: 12px;

                Text {
                    min-width: max(self.font-size * 8, min(self.preferred-width, self.max-width));
                    max-width: self.font-size * 50;
                    min-height: max(self.font-size * 3, self.preferred-height);
                    horizontal-alignment: left;
                    vertical-alignment: center;
                    wrap: word-wrap;
                    text: info;
                    font-weight: Typography.body.font-weight;
                    font-size: Typography.body.font-size;
                }
            }
        }
    }

**Comment by wuanzhuan** [comment]
when i set x,y of the scrollview, it is right. i think this is a bug.
        ScrollView {
            x: 0;
            y: 0;
            width: rect.preferred-width;
            height: rect.preferred-height;

            rect := Rectangle {
                background: Palette.background;
                border-color: Palette.border;
                border-width: 1phx;
                preferred-width: text-element.preferred-width + 12px;
                preferred-height: text-element.preferred-height + 12px;
                text-element := HorizontalLayout {
                    padding: 12px;
    
                    Text {
                        min-width: max(self.font-size * 8, min(self.preferred-width, self.max-width));
                        max-width: self.font-size * 50;
                        min-height: max(self.font-size * 3, self.preferred-height);
                        horizontal-alignment: left;
                        vertical-alignment: center;
                        wrap: word-wrap;
                        text: info;
                        font-weight: Typography.body.font-weight;
                        font-size: Typography.body.font-size;
                    }
                }
            }
        }

**Comment by ogoffart** [comment]
I think the problem might be that self.font-size is still zero before the popup is shown as we haven't set the parent window yet, and as a result, the size of the popup is not properly computed (as it needs to be computed before)

**Comment by ogoffart** [comment]
I tried to reproduce with This [slintpad link](https://slint.dev/releases/1.4.0/editor/?snippet=%0Aimport+%7B+AboutSlint%2C+Button%2C+VerticalBox+%7D+from+%22std-widgets.slint%22%3B%0Aexport+component+Demo+%7B%0A++++width%3A+200px%3B%0A++++height%3A+300px%3B%0A++++VerticalBox+%7B%0A++++++++alignment%3A+start%3B%0A++++++++Text+%7B%0A++++++++++++text%3A+%22Hello+World%21%22%3B%0A++++++++++++font-size%3A+24px%3B%0A++++++++++++horizontal-alignment%3A+center%3B%0A++++++++%7D%0A++++++++AboutSlint+%7B%0A++++++++++++preferred-height%3A+150px%3B%0A++++++++%7D%0A++++++++HorizontalLayout+%7B+alignment%3A+center%3B+Button+%7B+text%3A+%22OK%21%22%3B+clicked+%3D%3E+%7Bpopup.show%28%29%7D+%7D+%7D%0A++++%7D%0A%0A++++popup+%3A%3D+PopupWindow+%7B%0A++++++++x%3A+12px%3B%0A++++++++y%3A+12px%3B%0A%0A++++++++Rectangle+%7B%0A++++++++++++background%3A+blue%3B%0A++++++++++++border-color%3A+red%3B%0A++++++++++++border-width%3A+1phx%3B%0A++++++++++++preferred-width%3A+text-element.preferred-width+%2B+12px%3B%0A++++++++++++preferred-height%3A+text-element.preferred-height+%2B+12px%3B%0A++++++++++++text-element+%3A%3D+HorizontalLayout+%7B%0A++++++++++++++++padding%3A+12px%3B%0A%0A++++++++++++++++Text+%7B%0A++++++++++++++++++++min-width%3A+%7B%0A++++++++++++++++++++++++debug%28%22self.font-size%3A%22%2C+self.font-size%2C+%22preferred-width%3A%22%2C+self.preferred-width%29%3B%0A++++++++++++++++++++++++max%28self.font-size+*+8%2C+min%28self.preferred-width%2C+self.max-width%29%29%3B%0A++++++++++++++++++++%7D%0A++++++++++++++++++++max-width%3A+self.font-size+*+50%3B%0A++++++++++++++++++++min-height%3A+max%28self.font-size+*+3%2C+self.preferred-height%29%3B%0A++++++++++++++++++++horizontal-alignment%3A+left%3B%0A++++++++++++++++++++vertical-alignment%3A+center%3B%0A++++++++++++++++++++wrap%3A+word-wrap%3B%0A++++++++++++++++++++text%3A+%22mlkhj+jklkkjhl+lkjmlqds+fqsdfqsdfqfs+qsdf+qmjklm%22%3B%0A++++++++++++++++%7D%0A++++++++++++%7D%0A++++++++%7D%0A++++%7D%0A%7D%0A&style=native)


As one can see, the popup with is small.
But this is not a regression.

I think it is indeed a bug that self.font-size is zero. @tronical : do you agree?
@wuanzhuan is that the issue?

**Comment by wuanzhuan** [comment]
@ogoffart  the problem is caused by scrollview's backgroud and xy
![image](https://github.com/slint-ui/slint/assets/11628049/3ad6e411-84b6-4f47-9f2f-163f198af9a4)

the problem code is:
    popup := PopupWindow {
        y: root.height + 1rem;
        
        Rectangle {
            background: Palette.background;
            border-color: Palette.border;
            border-width: 1phx;  

            ScrollView {
                preferred-width: text-element.preferred-width + 12px;
                preferred-height: text-element.preferred-height + 12px;
                text-element := HorizontalLayout {
                    padding: 12px;
    
                    Text {
                        min-width: max(self.font-size * 8, min(self.preferred-width, self.max-width));
                        max-width: self.font-size * 50;
                        min-height: max(self.font-size * 3, self.preferred-height);
                        horizontal-alignment: left;
                        vertical-alignment: center;
                        wrap: word-wrap;
                        text: "hello world xxxxxxxxxxxxx!!!!!!!!!!!!!!!!!!!!";
                        //font-weight: Typography.body.font-weight;
                        font-size: 1rem;
                        color: red;
                    }
                }
            }
        }
    }

**Comment by ogoffart** [comment]
Could you paste a link to slintpad with the full code that can reproduce the issue?

**Comment by wuanzhuan** [comment]
@ogoffart sorry i  haven't used the slintpad. the full code as follows:
import { AboutSlint, Button, VerticalBox, Palette, ScrollView } from "std-widgets.slint";
export component Demo {
    width: 200px;
    height: 300px;
    VerticalBox {
        alignment: start;
        Text {
            text: "Hello World!";
            font-size: 24px;
            horizontal-alignment: center;
        }
        AboutSlint {
            preferred-height: 150px;
        }
        HorizontalLayout { alignment: center; Button { text: "OK!"; clicked => {popup.show()} } }
    }

    popup := PopupWindow {
        y: root.height + 1rem;
        
        Rectangle {
            background: Palette.background;
            border-color: Palette.border;
            border-width: 1phx;  

            ScrollView {
                preferred-width: text-element.preferred-width + 12px;
                preferred-height: text-element.preferred-height + 12px;
                text-element := HorizontalLayout {
                    padding: 12px;
    
                    Text {
                        min-width: max(self.font-size * 8, min(self.preferred-width, self.max-width));
                        max-width: self.font-size * 50;
                        min-height: max(self.font-size * 3, self.preferred-height);
                        horizontal-alignment: left;
                        vertical-alignment: center;
                        wrap: word-wrap;
                        text: "hello world xxxxxxxxxxxxx!!!!!!!!!!!!!!!!!!!!";
                        //font-weight: Typography.body.font-weight;
                        font-size: 1rem;
                        color: red;
                    }
                }
            }
        }
    }
}

**Comment by ogoffart** [comment]
I can reproduce the problem with a smaller testcase:

``
import { AboutSlint, Button, VerticalBox, ScrollView } from "std-widgets.slint";
export component Demo {
    width: 200px;
    height: 300px;

    Rectangle {
       width: r.preferred-width + 4px;
       height: r.preferred-height + 4px;
        
        r := Rectangle {
            background: blue;
            border-color: red;
            border-width: 1phx;  

            s := ScrollView {
                preferred-width: 50px;
                preferred-height: 50px;
            }
        }
    }
}
```

And indeed, the size is different with Slint 1.4 and 1.3 with the fluent style.  But the other style don't have this problem

**Comment by ogoffart** [comment]
Bisecting show it was caused by commit 95582e67948ff488c7fe3f5e6302eca4ca21defc

That impacted how the preferred size of parent of the ScrollView got computed.

**Comment by ogoffart** [comment]
looks like this was indeed fixed in 1.7.0
